src/chains/types.ts

```diff
@@ -1,6 +1,17 @@
 import { Chain as Metadata } from '@wagmi/chains';
 import { Address } from 'viem';
 
+export type SignatureType = {
+  prefixByte: number;
+  description: string;
+  // The data that is RLP encoded and signed to generate a signed transaction.
+  signedData: string[] | undefined;
+  // Some signature types are used to sign transactions, others are used to sign data.
+  signs: 'transaction' | 'data' | undefined;
+  references: string[];
+  notes?: string[];
+};
```

Comment:
Would include `hardforks: []string` here, protocol specs change. And then when importing L1 mainnet tx types in L2 support lists, filter by supported L1 forks. When L1 updates, not all L2s update immediately. E.g. 1559 wasn't on L2 everywhere the instant it was a mainnet L1 tx type.

Reply:
Yea I was thinking about this too. Since it applies to other things (e.g. opcodes) I'm going to add this more broadly in a future PR. Now tracking in https://github.com/mds1/evm-diff/issues/16